home/home_1.1/user_story_b.swift

```diff
@@ -0,0 +1,38 @@
+import Foundation
+
+func quadraticRoots(a: Double, b: Double, c: Double) -> (Double, Double)? {
+    let discriminant = b * b - 4 * a * c
+    if discriminant > 0 {
+        let root1 = (-b + sqrt(discriminant)) / (2 * a)
+        let root2 = (-b - sqrt(discriminant)) / (2 * a)
+        return (root1, root2)
+    } else if discriminant == 0 {
+        let root = -b / (2 * a)
+        return (root, root)
+    } else {
+        return nil
+    }
+}
+
+func main() {
+    print("Quadratic Equation Solver")
+    print("-------------------------")
+
+    print("Enter coefficients (a b c separated by a space):")
+    if let input = readLine() {
+        let coefficients = input.split(separator: " ").map(String.init)
```

Comment:
It is better to encapsulate this read logic in a separate function that will return optional triple of doubles. It will allow to make code more readable and decrease the count of nested if else statements.